filters/src/main/java/io/camunda/zeebe/process/test/filters/RecordStream.java

```diff
@@ -0,0 +1,128 @@
+package io.camunda.zeebe.process.test.filters;
+
+import io.camunda.zeebe.process.test.api.RecordStreamSource;
+import io.camunda.zeebe.protocol.record.Record;
+import io.camunda.zeebe.protocol.record.RecordValue;
+import io.camunda.zeebe.protocol.record.ValueType;
+import io.camunda.zeebe.protocol.record.value.DeploymentRecordValue;
+import io.camunda.zeebe.protocol.record.value.IncidentRecordValue;
+import io.camunda.zeebe.protocol.record.value.JobBatchRecordValue;
+import io.camunda.zeebe.protocol.record.value.JobRecordValue;
+import io.camunda.zeebe.protocol.record.value.MessageRecordValue;
+import io.camunda.zeebe.protocol.record.value.MessageStartEventSubscriptionRecordValue;
+import io.camunda.zeebe.protocol.record.value.MessageSubscriptionRecordValue;
+import io.camunda.zeebe.protocol.record.value.ProcessInstanceRecordValue;
+import io.camunda.zeebe.protocol.record.value.ProcessMessageSubscriptionRecordValue;
+import io.camunda.zeebe.protocol.record.value.TimerRecordValue;
+import io.camunda.zeebe.protocol.record.value.VariableDocumentRecordValue;
+import io.camunda.zeebe.protocol.record.value.VariableRecordValue;
+import io.camunda.zeebe.protocol.record.value.deployment.Process;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RecordStream {
+
+  private static final Logger LOG = LoggerFactory.getLogger("io.camunda.zeebe.process.test");
+
+  private RecordStreamSource recordStreamSource;
+
+  public RecordStream(final RecordStreamSource recordStreamSource) {
+    this.recordStreamSource = recordStreamSource;
+  }
```

Comment:
:wrench: We could change the visibility of the constructor to `private`. 

`RecordStream` has a static factory method `of(...)` that is used in the code. The constructor is not used except in the factory method.